nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -4,21 +4,54 @@
 import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.tree.AnnotatedTypeTree;
 import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.AssignmentTree;
+import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.ParameterizedTypeTree;
 import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
 import com.sun.tools.javac.code.Attribute;
 import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 
 /** Methods for performing checks related to generic types and nullability. */
 public final class GenericsChecks {
-
   private GenericsChecks() {
     // just utility methods
   }
 
+  @SuppressWarnings("UnusedVariable")
+  public static Type supertypeMatchingLHS(
+      Type.ClassType lhsType, Type.ClassType rhsType, VisitorState state) {
+    // interfaces
+    List<Type> listOfDirectSuperTypes = rhsType.all_interfaces_field;
```

Comment:
We can clean up this code a bunch by calling `state.getTypes.closure(rhsType)`, which will give you a list of all supertypes, including both superclasses and interfaces.